solver/src/http_solver.rs

```diff
@@ -114,140 +155,111 @@ impl HttpSolver {
                     fee: 0.003,
                     mandatory: false,
                 };
-                (index.to_string(), uniswap)
+                (index.clone(), uniswap)
             })
-            .collect())
-    }
-
-    async fn create_body(&self, orders: &[Liquidity]) -> Result<BatchAuctionModel> {
-        Ok(BatchAuctionModel {
-            tokens: self.tokens(orders),
-            orders: self.orders(orders),
-            uniswaps: self.uniswaps(orders).await?,
-            ref_token: self.token_to_string(&H160::zero()),
-            default_fee: 0.0,
-        })
+            .collect()
     }
-}
 
-#[async_trait::async_trait]
-impl Solver for HttpSolver {
-    async fn solve(&self, orders: Vec<Liquidity>) -> Result<Option<Settlement>> {
+    async fn send(&self, model: &BatchAuctionModel) -> Result<SettledBatchAuctionModel> {
         let mut url = self.base.clone();
         url.set_path("/solve");
         self.config.add_to_query(&mut url);
-        let body = self.create_body(orders.as_slice()).await?;
+        let query = url.query().map(ToString::to_string).unwrap_or_default();
         let mut request = self.client.post(url);
         if let Some(api_key) = &self.api_key {
             request = request.header("X-API-KEY", api_key);
         }
-        let request = request.json(&body);
+        let body = serde_json::to_string(&model).context("failed to encode body")?;
+        tracing::debug!("request query {} body {}", query, body);
+        let request = request.body(body);
 
-        let response = request
-            .send()
-            .await
-            .context("failed to send solver request")?
-            .error_for_status()
-            .context("solver response was unsuccessful")?;
-        let body = response
+        let response = request.send().await.context("failed to send request")?;
+        let status = response.status();
+        let body_bytes = response
             .bytes()
             .await
             .context("failed to get response body")?;
-        let _decoded: Solution =
-            serde_json::from_slice(&body).with_context(|| match std::str::from_utf8(&body) {
-                Ok(body) => format!("failed to decode response body: {}", body),
-                Err(_) => format!("failed to decode response body: {:?}", body),
-            })?;
-        Ok(None)
+        let body_str =
+            std::str::from_utf8(body_bytes.as_ref()).context("failed to decode response body")?;
+        tracing::debug!("response body {}", body_str);
```

Comment:
added more logging here to ensure we always log the response even on non success status code